kse/src/main/java/org/kse/gui/actions/ExamineClipboardAction.java

```diff
@@ -270,36 +273,42 @@ private void showPrivateKey(byte[] data, CryptoFileType fileType) throws IOExcep
                 return;
             }
             privKey = Pkcs8Util.loadEncrypted(data, password);
+            format = PrivateKeyFormat.PKCS8;
             break;
         case UNENC_PKCS8_PVK:
             privKey = Pkcs8Util.load(data);
+            format = PrivateKeyFormat.PKCS8;
             break;
         case ENC_OPENSSL_PVK:
             password = getPassword();
             if (password == null || password.isNulled()) {
                 return;
             }
             privKey = OpenSslPvkUtil.loadEncrypted(data, password);
+            format = PrivateKeyFormat.PKCS1;
             break;
         case UNENC_OPENSSL_PVK:
             privKey = OpenSslPvkUtil.load(data);
+            format = PrivateKeyFormat.PKCS1;
             break;
         case ENC_MS_PVK:
             password = getPassword();
             if (password == null || password.isNulled()) {
                 return;
             }
             privKey = MsPvkUtil.loadEncrypted(data, password);
+            format = PrivateKeyFormat.MSPVK;
             break;
         case UNENC_MS_PVK:
             privKey = MsPvkUtil.load(data);
+            format = PrivateKeyFormat.MSPVK;
             break;
         default:
             break;
         }
 
         DViewPrivateKey dViewPrivateKey = new DViewPrivateKey(frame, res.getString(
-                "ExamineClipboardAction.PrivateKeyDetails.Title"), "", privKey, applicationSettings);
+                "ExamineClipboardAction.PrivateKeyDetails.Title"), "", privKey, applicationSettings, Optional.ofNullable(format));
```

Comment:
Normally the use of Optional as a method parameter is discouraged in Java. In this case we have an enum, so another value like UNDEFINED might have been another option. That being said, I like your solution, it makes the code in DViewPrivateKey quite concise and elegant.